### DISCUSSION TITLE
Style template with colour variants

### DISCUSSION THREAD

**Discussion by Quin452** [tailwindlabs/tailwindcss#3767] [Help]
I'm sorry if this has been asked before, but TBH, I'm not entirely too sure what to search for.

Basically I've been creating my own classes using the @apply rule.  So, I create a basic `.btn` class (to use on buttons), and then have the colour variants `.btn-success { @apply btn bg-green-600; }`  Now, this works well, but I wondered if there is a way to make this better.

Ideally some sort of loop, so I can make .btn-success, .btn-error, .btn-warning all in some sort of loop, rather than copying and pasting each time.
So if I want to change the colour strength (?, the number), I can do it from 1 line rather than find each instance.
So change -600 to -400 once.

My ideal would be `@apply btn bg-{color}-600 border-{color}-600;`

**Comment by dance2die** [comment]
You can extend colors adding such dynamic colors in the configuration like `success/warning/error`,

__Demo__: https://play.tailwindcss.com/AcBTIwf7Du

```js
const colors = require('tailwindcss/colors')

module.exports = {
  theme: {
    extend: {
      colors: {
        'light-blue': colors.lightBlue,
        cyan: colors.cyan,
        success: colors.green['600'],
        warning: colors.yellow['400'],
        error: colors.red['600'],
      },
    },
  },
}
```

 and then `@apply` them in your CSS.

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .btn {
    @apply w-full h-full p-8 text-white;
  }

  .btn-success {
    @apply btn bg-success border-success;
  }

  .btn-warning {
    @apply btn bg-warning border-warning;
  }

  .btn-error {
    @apply btn bg-error border-error;
  }
}
```

Later on, when you need to change the colors, you can modify your configuration to apply the change globally. 

If you use happen to do this often, you can create a preset or a plugin, publish to NPM and include in your project :)

___

**related question (answered)**: https://github.com/tailwindlabs/tailwindcss/discussions/3785

**Comment by simonswiss** [comment]
Hey!

You can do loops in your CSS if you're using a pre-processor like [Sass](https://sass-lang.com/documentation/at-rules/control/each), or you could use this [PostCSS plugin](https://github.com/antyakushev/postcss-for).

Hope that gives you some direction to get started 👍
